packages/flame/lib/src/sprite_batch.dart

```diff
@@ -180,18 +183,36 @@ class SpriteBatch {
     );
   }
 
+  static const String _generatedAtlasKeySuffix = '#withFlipAttached';
+
   static Future<Image> _generateAtlas(Images? images, String path) async {
```

Comment:
Have you made sure that this one isn't called more than necessary? I don't really understand why the `PictureRectorder` weren't able to handle this usecase... In which cases does it need to be re-generated?

Reply:
How I replicated the performance issues mentioned in #1614 was to load a test .tmx file I had in a for loop x30 because I assumed my test map was just not complicated enough to see the issue.
```
for (int i = 0; i < 30; i++) {
      TiledComponent anotherMap =
          await TiledComponent.load('map.tmx', Vector2.all(64));
      add(anotherMap);
```
Running the above code against [`60df3b`](https://github.com/flame-engine/flame/commit/60df3b9f60f538fbad7a3d806f5d38262ab6d66c) the game runs normally.

I set a breakpoint within `_generateAtlas` to test whether this function was being called more than necessary (or worse per game tick) based on st-pasha's [comment](https://github.com/flame-engine/flame/issues/1614#issuecomment-1123076693). Once the spriteBatch has been loaded images are not recreated - `_generateAtlas` is not called again after the initial load, but performance in the game is terrible.

After replacing the `PictureRecorder` code performance was the same as [`60df3b`](https://github.com/flame-engine/flame/commit/60df3b9f60f538fbad7a3d806f5d38262ab6d66c).

I can't actually confirm that `PictureRecorder` is the culprit here since I replaced the whole code block with something new. Something in the current approach is causing a performance problem however.